sullivan_a07b.js

```diff
@@ -0,0 +1,90 @@
+/*  Anna Sullivan
+    sullivan_a07b.js
+    INFO2124WW
+    Thoendel
+    2/1/2020
+*/
+/* DO NOT MODIFY CODE BETWEEN THESE LINES */
+const standardInput = process.stdin;            
+standardInput.resume();
+standardInput.setEncoding('utf8');
+const badFoods = [];
+const groceryItems = [];
+clearScreen();
+console.log(getPrompt(groceryItems));
+/* DO NOT MODIFY CODE BETWEEN THESE LINES */
+
+//Step 1
+buildFoodsList (badFoods, "bread", "beer", "cheese", "milk", "cookie");     /*added 3 more string values as arguments*/   
+
+standardInput.on('data', function (text) {
+    //This line will remove line breaks 
+    //\n\r in window, \r in macOS
+    text = text.replace('\n', '').replace('\r', '');
+    //Step 4 Begin
+
+    /*if else to determine the user input using decision stucture*/
+if((text == "q") || (text == "Q")){
+    clearScreen();
+    console.log(`Bye!`);
+    process.exit();
+}else{
+    clearScreen();
+    checkItem(text, badFoods);
+    if(checkItem() == false){
+        text += formatItem();
+        let addItem = `Item added to list!`;
+        return addItem;
+    }else if(checkItem() == true){
+        let error = `Error: that item is not safe for your allergies.\nIt has not been added to your list. ${getPrompt()}`
+        return error;
+    }
+}
+    
+    //Step 4 End
+});
+
+/* DEFINE YOUR FUNCTIONS BETWEEN THIS LINE */
+
+//Step 2
+function buildFoodsList (badFoods, ...foods){
+    let total = 0;
+    for(let food of foods){
+        total += food;

```

Comment:
if total is supposed to be an array - we will want to push or unshift those foods in and when we declare total it should be something like:
let total = [];

Also - we could just push those foods into badFoods since it is a global array. That's how I'd go.

Reply:
for(let food of foods){
badFoods.push(food); 
}
like that?
or food.push(badFoods);
I was just very confused how to use the methods with the arrays.

Reply:
yes! since you want the items to be in badFoods - you want to use the push method of that array. That's why its badFoods.push(food)
Remember - I'm here to help with that sort of confusion - should you have any future questions 😄